File: src/search/elasticSearch.go
package search

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
	"strconv"
)



type GetResponse struct {
	Took     int  `json:"took"`
	TimedOut bool `json:"timed_out"`
	Shards   struct {
		Total      int `json:"total"`
		Successful int `json:"successful"`
		Skipped    int `json:"skipped"`
		Failed     int `json:"failed"`
	} `json:"_shards"`
	Hits struct {
		Total struct {
			Value    int    `json:"value"`
			Relation string `json:"relation"`
		} `json:"total"`
		MaxScore float64 `json:"max_score"`
		Hits     []struct {
			Index  string  `json:"_index"`
			Type   string  `json:"_type"`
			ID     string  `json:"_id"`
			Score  float64 `json:"_score"`
			Source struct {
				Pref       string `json:"pref"`
				Employment string `json:"employment"`
				Line       struct {
					Name    string   `json:"name"`
					Station []string `json:"station"`
				} `json:"line"`
			} `json:"_source"`
		} `json:"hits"`
	} `json:"hits"`
}

type GetBody struct {
	Query struct {
		Term struct {
			ID string `json:"_id"`
		} `json:"term"`
	} `json:"query"`
}

func Get(id int) string {
	url := "http://elasticsearch:9200/search_job/_search"
	//url := "http://127.0.0.1:9200/search_job/_search"
	var getBody = GetBody{}
	getBody.Query.Term.ID = strconv.Itoa(id)
	v, _ := json.Marshal(getBody)
	req, _ := http.NewRequest("GET", url, bytes.NewBuffer(v))
	req.Header.Set("Content-Type", "application/json")
	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		fmt.Println(err)
	}

	body, _ := ioutil.ReadAll(resp.Body)
	fmt.Println(body)
	fmt.Println(string(body))
	var getResponse GetResponse
	unmarshalErr := json.Unmarshal(body, &getResponse)
	if unmarshalErr != nil {
		fmt.Println(unmarshalErr)
	}

	for _, hit := range getResponse.Hits.Hits {
		fmt.Println("id", hit.ID)
		fmt.Println("pref", hit.Source.Pref)
		fmt.Println("employment", hit.Source.Employment)
		fmt.Println("employment", hit.Source.Line.Name)
		for _, station := range hit.Source.Line.Station {
			fmt.Println("station", station)
		}

	}

	fmt.Println(getResponse.Hits.Hits)

	return "ok"
}

type Job struct {
	Pref string `json:"pref"`
	Employment string `json:"employment"`
	Line []Line `json:"line"`
}

type Line struct {
	Name string `json:"name"`
	Station []Station `json:"station"`
}

type Station struct {
	Name string `json:"name"`
}

func Make(id int) {
	url := "http://172.54.0.54:9200/search_job/_doc/"
	//url := "http://127.0.0.1:9200/search_job/_doc/"
	url = url + strconv.Itoa(id)
	job := Job{
		"東京" + strconv.Itoa(id),
		"full-time",
		[]Line{
			{"山手線",
					[]Station {
						{"東京"},
					},
			},
		},
	}
	v, _ := json.Marshal(job)
	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(v))
	req.Header.Set("Content-Type", "application/json")
	client := &http.Client{}
	_, err := client.Do(req)
	if err != nil {
		fmt.Println(err)
	}
}

File: src/ElasticSearch.php
<?php


class ElasticSearch
{

    public function get()
    {
        $ch = curl_init("http://elasticsearch:9200/search_job/_search");
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "GET");
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);

        for ($i = 0; $i < 100; $i++ ) {
            echo $i;
            curl_exec($ch);
        }
        $response = curl_exec($ch);
        var_dump($response);

        curl_close($ch);
    }

    public function make(int $id)
    {
        $ch = curl_init("http://elasticsearch:9200/search_job/_doc/" . $id);
        curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($this->station($id)));
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
        curl_setopt($ch, CURLOPT_HTTPHEADER, ['Content-Type: application/json;  charset=UTF-8']);
        curl_exec($ch);
        curl_close($ch);
    }

    public function station(int $id)
    {
        return [
            'pref' => '東京' . $id,
            'employment' => 'full-time',
            'line' => [
                'name' => '山手線',
                'station' => [
                    '東京'
                ]
            ]
        ];
    }
}
echo 'start';

if (!$hoge = 0) {
    var_dump($hoge);
}

var_dump($hoge);

return;

$start = microtime(true);
$es = new ElasticSearch();
$es->get();
//$es->make(2);
//for ($i = 0; $i < 100; $i++) {
//    $es->make($i);
//}
var_dump(microtime(true) - $start);